### ISSUE TITLE
[REC] Balance opener changes from Wrath, Wrath, Starfire

### ISSUE THREAD

**Issue by noiseuu** [Hekili/hekili#1178]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this bug report (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

Hello

I just tried addon, and i saw one thing
In Dreamgrove it suggest opener: Night Fae - no BoAT (with NB):
Wrath, Wrath, Starfire, dots, CA, Starsurge 2x, Convoke, Starsurge until you have no more AsP, New Moon, Half Moon, Starsurge, Full Moon

But hekili wants use Wrath, Dots, CA, Starsurge 2x, Convoke
And im atm talking about missing that starfire

### How to Reproduce

1.  Target dummy
2. Talents for single target
3. Followed my full opener

### Snapshot (Link)

https://pastebin.com/KxP1eZc1

### Raidbots Sim Report (Link)

https://gyazo.com/5414ba51f0f755af8cf0bfda34867a27

**Comment by Hekili** [comment]
The sim opener is contingent on you standing far enough to start your second Wrath before your first hits the target.

**Comment by noiseuu** [comment]
But why it doesnt suggest after 2x Wrath 1x Starfire? Is because its precombat thing?

**Comment by Hekili** [comment]
Take a snapshot for that scenario.

**Comment by noiseuu** [comment]
https://pastebin.com/k592Kb6U
This pastebin is atm max range from target

**Comment by Hekili** [comment]
But it's not a fresh opener.

**Comment by noiseuu** [comment]
Sorry, my bad

https://pastebin.com/3cHzC15n
This is max range, i hope this gives more information. I sometimes doesnt know make snapshot :D
And this pastebin i was having 50 astral power

**Comment by mickeygeecom** [comment]
In extension to this Venthyr balance is not recommending covenant and/or Celestial Alignment (also Incarnation, talent) on opener.On Sun, Jul 25, 2021 at 6:43am, noiseuu ***@***.***> wrote:
Sorry, my bad
https://pastebin.com/3cHzC15n
This is max range, i hope this gives more information. I sometimes doesnt know make snapshot :D

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/Hekili/hekili/issues/1178#issuecomment-886145946",
"url": "https://github.com/Hekili/hekili/issues/1178#issuecomment-886145946",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by AleSSouza** [comment]
I can confirm the issue regarding hekili not recommending CA/INC/Ravenous Frenzy.
Can u check, pelase?

Don't know if I did the snapshot right but here it goes.

[https://pastebin.com/NkxQyZkX](url)


> In extension to this Venthyr balance is not recommending covenant and/or Celestial Alignment (also Incarnation, talent) on opener.On Sun, Jul 25, 2021 at 6:43am, noiseuu ***@***.***> wrote: Sorry, my bad https://pastebin.com/3cHzC15n This is max range, i hope this gives more information. I sometimes doesnt know make snapshot :D —You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe. [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "[#1178 (comment)](https://github.com/Hekili/hekili/issues/1178#issuecomment-886145946)", "url": "[#1178 (comment)](https://github.com/Hekili/hekili/issues/1178#issuecomment-886145946)", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by Hekili** [comment]
Open a separate ticket for the cooldowns issue.

For the opener, supply a snapshot of the opener. The snapshots I've looked at here are not taken at the opener. For example, you've already used CA/Inc.  if you're reporting that the addon doesn't recommend Wrath, Wrath, Starfire to open, supply a snapshot of what it does recommend when you're opening from max range.

**Comment by noiseuu** [comment]
https://gyazo.com/2aaed5fdd5b55699a810e54766194380
i took this gif, i can still give u pastebin, but i feel u dont understand when i post it. Every pastebin is with cooldowns and max range so.

**Comment by Hekili** [comment]
* pastebin.com/k592Kb6U
  - This snapshot has Incarnation up, IQD's buff up, and the target already has Moonfire, Sunfire, and Stellar Flare applied.  This is not a snapshot from the circumstances you were reporting.
* pastebin.com/3cHzC15n
  - Same as above (but earlier in Incarnation).
* pastebin.com/NkxQyZkX
  - This is at the tail end of a Lunar Eclipse.

A snapshot captures a point in time.  You snapshot when it is recommending what you're concerned about, not after.

* gyazo.com/2aaed5fdd5b55699a810e54766194380

The GIF is of the correct timeframe, but doesn't offer any useful debugging information, because it's not a snapshot.  Snapshot the opener recommendations (Wrath, Wrath, Starfire).  Snapshot when the recommendations change from Wrath, Wrath, Starfire.

The result is mainly going to be that combat started, but I'll answer when there's data I can analyze.

**Comment by noiseuu** [comment]
https://pastebin.com/5vuYe1jD
Okay lets try that, if i fucked up then i cant use this your snapshot.
And i just realise i dont have correct talents, but i feel same opener with those talents is correct

**Comment by Hekili** [comment]
Well, I did ask for *two* snapshots, and that is one single snapshot, so we're not off to a great start.

From the look of it, I'm assuming that this is after you've cast Wrath once, since you have 56 AP (generated 6 from your Wrath cast) and Wrath will hit in 0.68 seconds.

At the point you took the snapshot, you're in-combat, so the remainder of the `precombat` priority is no longer applicable.  Any thoughts on why you're in combat before your Wrath has impacted?

**Comment by Hekili** [comment]
I'm also confused why your GIF shows 3 icons but your snapshot shows only 1.

**Comment by Hekili** [comment]
@mickeygeecom @AleSSouza Reminder to open a ticket for the cooldowns issue.

**Comment by noiseuu** [comment]
> 
> 
> I'm also confused why your GIF shows 3 icons but your snapshot shows only 1.

I enjoy more have 1 icon, than 3 and i just showed how icons was.

**Comment by noiseuu** [comment]
> 
> 
> Well, I did ask for _two_ snapshots, and that is one single snapshot, so we're not off to a great start.
> 
> From the look of it, I'm assuming that this is after you've cast Wrath once, since you have 56 AP (generated 6 from your Wrath cast) and Wrath will hit in 0.68 seconds.
> 
> At the point you took the snapshot, you're in-combat, so the remainder of the `precombat` priority is no longer applicable. Any thoughts on why you're in combat before your Wrath has impacted?

So, u want precombat snapshot not like in combat?

**Comment by Hekili** [comment]
> Snapshot the opener recommendations (Wrath, Wrath, Starfire).
> Snapshot when the recommendations change from Wrath, Wrath, Starfire.

**Comment by noiseuu** [comment]
https://pastebin.com/SUtAXqFb ( snapshot the opener recommendations )
https://pastebin.com/HUdPgHn8 ( snapshot when recommendations change )

**Comment by Hekili** [comment]
So the issue centers on the simulation not treating you as though you're in-combat until your first damaging spell impacts.  However, in-game, you are in combat as soon as Wrath finishes casting.  I can modify the way the opener is implemented and probably take care of this, but I'll also need to wait on the other issue report about Venthyr and CDs.

In the meantime, use your good judgment and open with Wrath, Wrath, Starfire and pick up from there, if you're starting from a "fresh" pull.

**Comment by noiseuu** [comment]
np, sorry taking your time. now i understand what did u mean.

**Comment by Hekili** [comment]
No worries, it was a good call-out.  I'm not sure why the player goes in-combat before the target's actually been hit.  May just be something with dummies.  Not sure.

**Comment by Hekili** [comment]
Should be resolved in 1.0.17.
